ludicrousdb/includes/class-ludicrousdb.php

```diff
@@ -1303,7 +1303,8 @@ protected function single_db_connect( $dbhname, $host, $user, $password ) {
 					$socket = substr( $maybe_socket, 1 );
 				}
 			} else {
-				$socket = $port_or_socket;
+				// otherwise it will fail if the host is specified originally like ':/var/lib/mysql/mysql.sock', since we will have a port appended here still
+				$socket = strtok( $port_or_socket, ':' );
```

Comment:
This will trigger on normal hosts `(127.0.0.1:3306)`, bracketed IPv6 `([::1]:3306)`, or hostnames with ports.

That doesn't seem right.

What about:
```php
strncmp($host, ':/', 2) === 0
```

or

```php
preg_match('#^unix://#', $host)
```